ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt

```diff
@@ -37,13 +39,14 @@ internal interface PaywallViewModel {
 }
 
 internal class PaywallViewModelImpl(
-    offering: Offering?,
+    private val applicationContext: WeakReference<Context>,
```

Comment:
This is a _huge_ dependency, especially consider all we want is the application name.
Something that would make this a lot simpler and easier to test is create an interface similar to `VariableDataProvider`. Then we can easily mock it in previews / snapshots, for example.

Reply:
Mentioned [here](https://github.com/RevenueCat/purchases-android/pull/1251#discussion_r1328378217) but I think we can't really create an interface for this like we would do in swift.